core/templates/services/messenger.service.ts

```diff
@@ -19,147 +19,201 @@
  * be attempted due to cross-domain security issues.)
  */
 
-angular.module('oppia').factory('MessengerService', [
-  '$log', '$window', function($log, $window) {
-    var isPositiveInteger = function(n) {
-      return (typeof n === 'number' && n % 1 === 0 && n > 0);
-    };
-    var isBoolean = function(b) {
-      return typeof b === 'boolean';
-    };
-
-    var SUPPORTED_HASHDICT_VERSIONS = [
-      '0.0.0', '0.0.1', '0.0.2', '0.0.3'
-    ];
-
-    var MESSAGE_VALIDATORS = {
-      heightChange: function(payload) {
-        return isPositiveInteger(payload.height) && isBoolean(payload.scroll);
-      },
-      explorationLoaded: function() {
-        return true;
-      },
-      stateTransition: function(payload) {
-        return Boolean(payload.oldStateName) || Boolean(payload.newStateName);
-      },
-      explorationReset: function(payload) {
-        return Boolean(payload.stateName);
-      },
-      explorationCompleted: function() {
-        return true;
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { LoggerService } from 'services/contextual/logger.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+interface MessageValidatorsType {
+  heightChange: ((payload: {height: number; scroll: number}) => boolean);
+  explorationLoaded: (() => boolean);
+  stateTransition: (
+    (payload: {oldStateName: string; newStateName: string}) => boolean);
+  explorationReset: ((payload: {stateName: string}) => boolean);
+  explorationCompleted: (() => boolean);
+}
+
+interface HeightChangeData {
+  height: number;
+  scroll: number;
+}
+
+interface ExplorationLoadedData {
+  explorationVersion: number;
+  explorationTitle: string;
+}
+
+interface StateTransitionData {
+  explorationVersion: number;
+  oldStateName: string;
+  jsonAnswer: string;
+  newStateName: string;
+}
+
+interface ExplorationCompletedData {
+  explorationVersion: number;
+}
+
+interface GetPayloadType {
```

Comment:
Seems this should actually be a `Map` with pre-populated keys and values?

Reply:
Results in type errors since we are defining functions actually.